fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/Binder.java

```diff
@@ -162,6 +153,76 @@ private GroupExpression extractGroupExpression(Pattern pattern, Group group) {
         }
     }
 
+    /**
+     * Expression binding for MULTI_JOIN, which contains only JOIN/SCAN nodes
+     * <p>
+     * NOTE: Why not match using regular pattern ?
+     * 1. Regular matching cannot bind the tree but only the partial expression. But MULTI_JOIN needs to extract
+     * the entire tree
+     * 2. Regular matching is extremely slow in this case since it needs to recursive descent the tree, build the
+     * binding state and check the expression at the same time. But MULTI_JOIN could enumerate the GE without any check
+     */
+    class MultiJoinBinder {
```

Comment:
1. move it to MultiJoinBinder.java
2. extends it from Binder?

Reply:
it shares the state with `Binder` and may be invoked during the normal binding procedure, in the case that try to bind `AGG -> MULTI_JOIN`.